buildpacks/ruby/src/layers/ruby_install_layer.rs

```diff
@@ -89,8 +157,8 @@ impl<'a> Layer for RubyInstallLayer<'a> {
 
                 Ok(ExistingLayerStrategy::Keep)
             }
-            Changed::Stack(_old, _now) => {
-                log_step(format!("Clearing cache {}", fmt::details("stack changed")));
+            Changed::Target(_old, _now) => {
+                log_step(format!("Clearing cache {}", fmt::details("OS changed")));
```

Comment:
Would it perhaps make sense to use "Platform" instead of "OS" here? Since the change could be caused by either a change in the distro and arch? Or alternatively writing comparing the individual TargetId fields and logging explicitly what changed (as "platform" can be kinda vague) (e.g. "OS/distro changed", "Arch changed")?

Reply:
Short: I agree. I will rev this to emit distro version and arch changing as separate warnings. 

Long (talking out loud):
I noticed that too. I'm not sure. Ultimately I'm not really happy with the way I'm doing these messages, and I want to rev them but I also want to wait until https://github.com/heroku/libcnb.rs/pull/814 lands so I can do some more automation such as https://github.com/heroku/buildpacks-ruby/pull/203/files#diff-1cedf1bfbafd5ca2b85e8261de347a2318ab3ea87abe49eeff117fb3f380faa9R147-R157. I like what Ed has done with listing out ALL of the changes as someone could change ruby version and OS version at the same time.

Ultimately a lot of Ruby users aren't super into the weeds in the OS space. I think if I said "platform" they would have no clue what that is. "Isn't Heroku my platform." I wanted to pick wording that **feels** correct even if it's not the most detailed explanation for what's actually happened. For Ruby users "Distro" is likely more clearly associated with operating systems. 

Realistically though the main cache invalidation reasons people will see are due to:

- Distro version changing (22.04 to 24.04)
- Arch changing (amd64 to arm64)

The second case (arch changing) isn't technically implemented yet in this PR. I could make the case of "we will cross that bridge later." But, I do want to re-consider how I'm storing the data.

In https://github.com/heroku/buildpacks-ruby/pull/203/files#diff-1cedf1bfbafd5ca2b85e8261de347a2318ab3ea87abe49eeff117fb3f380faa9R147-R157 I'm using metadata keys to auto generate a message and if I tried something like that now it would say something like "target_id changed" which is not really helpful. I think it makes sense to either:

- Tell serde to flatten the struct storage so there is no high level `target_id` key and instead the user gets "distro_version changed"
- Ditch the struct and go with what Ed did in the python buildpack of storing only the keys needed directly. 

I feel the second option is more "clear." I think I would like to keep the struct for passing around the data and centralizing knowledge of the buildpack in regards to Target information. However, I would like to switch to using explicit cache keys.

So, that's a long way to say...I agree.

I might also do a little work to store and emit multiple changes like the python buildpack. Though that's increasing the scope of the PR. To that end, I think handling them explicitly as single enums is good, and a followup or possibly stacked PR to instead store them as a vector of enums is a better way to go.

Reply:
I think this all makes a lot of sense -- and yes since you're not supporting more than one arch at this point "OS Changed" seems fine for now (especially to avoid increasing the scope of this PR)! Just wanted to make sure it was on your radar when you implement multi-arch stuff

Reply:
I'll change this back to draft while I work on making that change, I'll update when I'm ready for a re-review.

In the meantime, if you have any other comments, feel free to leave them.